presto-tests/src/test/java/com/facebook/presto/resourcemanager/TestDistributedTaskInfoResource.java

```diff
@@ -0,0 +1,114 @@
+import static com.facebook.presto.utils.QueryExecutionClientUtil.getTaskInfo;
+import static com.facebook.presto.utils.QueryExecutionClientUtil.runToCompletion;
+import static com.facebook.presto.utils.QueryExecutionClientUtil.runToFirstResult;
+import static com.facebook.presto.utils.ResourceUtils.getResourceFilePath;
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+import static java.lang.Thread.sleep;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public class TestDistributedTaskInfoResource
+{
+    private static final int COORDINATOR_COUNT = 2;
+    private HttpClient client;
+    private TestingPrestoServer coordinator1;
+    private TestingPrestoServer coordinator2;
+    private TestingPrestoServer resourceManager;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        client = new JettyHttpClient();
+        DistributedQueryRunner runner = createQueryRunner(ImmutableMap.of("query.client.timeout", "20s"), COORDINATOR_COUNT);
+        coordinator1 = runner.getCoordinator(0);
+        coordinator2 = runner.getCoordinator(1);
+        Optional<TestingPrestoServer> resourceManager = runner.getResourceManager();
+        checkState(resourceManager.isPresent(), "resource manager not present");
+        this.resourceManager = resourceManager.get();
+        coordinator1.getResourceGroupManager().get().addConfigurationManagerFactory(new FileResourceGroupConfigurationManagerFactory());
+        coordinator1.getResourceGroupManager().get()
+                .setConfigurationManager("file", ImmutableMap.of("resource-groups.config-file", getResourceFilePath("resource_groups_config_simple.json")));
+        coordinator2.getResourceGroupManager().get().addConfigurationManagerFactory(new FileResourceGroupConfigurationManagerFactory());
+        coordinator2.getResourceGroupManager().get()
+                .setConfigurationManager("file", ImmutableMap.of("resource-groups.config-file", getResourceFilePath("resource_groups_config_simple.json")));
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown()
+    {
+        closeQuietly(coordinator1);
+        closeQuietly(coordinator2);
+        closeQuietly(resourceManager);
+        closeQuietly(client);
+        coordinator1 = null;
+        coordinator2 = null;
+        resourceManager = null;
+        client = null;
+    }
+
+    @Test(timeOut = 220_000)
+    public void testDistributedGetTaskInfo()
+            throws Exception
+    {
+        runToCompletion(client, coordinator1, "SELECT 1");
+        runToFirstResult(client, coordinator1, "SELECT * from tpch.sf101.orders");
+        sleep(SECONDS.toMillis(5));
+        waitUntilCoordinatorsDiscoveredHealthyInRM(SECONDS.toMillis(15));
+        for (TaskInfo actualTaskInfo : coordinator2.getTaskManager().getAllTaskInfo()) {
```

Comment:
why are we pulling the getAllTaskInfo from coordinator2 here? Just pull it from coordinator 1 and try to fetch the details from coordinator 2, is a good test.